omegaconf/dictconfig.py

```diff
@@ -189,60 +162,108 @@ def _validate_set_merge_impl(self, key: Any, value: Any, is_assign: bool) -> Non
         else:
             target = self._get_node(key)
 
-        if target is None:
-            return
+        def is_value_node(value: Any) -> bool:
+            valid = False
+            if isinstance(value, ValueNode) and self._has_element_type():
+                valid = True
+            return valid
```

Comment:
You can replace this function with a single line where you use it:
```
is_value_node = isinstance(value, ValueNode) and self._has_element_type()
```

Reply:
This also reads wrong, if self has or has no element has no bearing on the value being a value node.